vaadin-components-gwt/src/main/java/com/vaadin/components/grid/GridComponent.java

```diff
@@ -439,7 +443,7 @@ public void sort(SortEvent<Object> event) {
             jsSort.push(sortOrder);
         }
         $(container).trigger("sort");
-        refresh();
+        refresh(false);
```

Comment:
Sorting never changes the number of items in the datasource and doesn't require any size recalculation so refresh/redraw shouldn't be needed here at all. A call to getDataSource().clearCache(null); instead should be enough.